### ISSUE TITLE
You can run an executable by changing its ending to .zip and then double-clicking it.

### ISSUE THREAD

**Issue by Binekrasik** [files-community/Files#14137]
### Description

You can rename an .exe to .zip and when you double-click it, it runs the executable. Windows file explorer doesn't do this.

### Steps To Reproduce

1. Get some executable
2. Change its ending to .zip
3. Double-click the "zip file"
4. The executable runs

### Requirements

It should display an error that the archive cannot be opened, like windows file explorer.

### Files Version

2.5.31.0

### Windows Version

10.0.22635.2776

### Log File

[debug.log](https://github.com/files-community/Files/files/13534030/debug.log)

**Comment by Josh65-2201** [comment]
Thanks for the report, please update to 3.0.8 and try reproduce this. https://files.community/download

**Comment by Binekrasik** [comment]
Yes, the issue still persists in the new version of Files ( 3.0.14.0 ).

**Comment by Josh65-2201** [comment]
I'm unable to reproduce this. I have tried `.exe.zip` and just `.zip` both showing the error message and not opening the exe. What the full file name of the one causing this?

**Comment by Binekrasik** [comment]
The full name is windowkill-vulkan.zip (tried this on a portable godot game), but it seems it cannot be reproduced on other executables. It also works with other endings, like .rar and .7z.

**Comment by Binekrasik** [comment]
I don't think so. Someone can, for example, send you a virus that pretends to be a zip file.

**Comment by yaira2** [comment]
I see your point, but that aside, I can't repro this issue. Do you have archive files set to open in Files?

**Comment by Binekrasik** [comment]
No, I don't, but tried it and didn't see any difference. However, I've tried to open the 'zip' with Files in file explorer, and that threw an error saying 'NotAFolder', whereas if you try it in Files, it will run the executable without any error.

**Comment by yaira2** [comment]
Did you set it to always open in Files?

**Comment by yaira2** [comment]
I believe you but I must be a missing a step when trying to repro this, can you upload a screen recording (you can use the attachfile feature in GitHub) so I can make sure I'm following the same steps?

**Comment by MartinGamesCZ** [comment]
Hi, I have been able to reproduce the issue.

For certain exe files it works. You change the file extension from .exe to .zip and it still opens when you double click.

My version: 3.0.8.0 on Windows 11 Home 23H2 build 22631.2715

I only tested it for the same exe as @Binekrasik, the main executable of [this game](https://torcado.itch.io/windowkill) - You can download it for free in case you want to test the bug (the money is just optional donation)

Here is a video:

https://github.com/files-community/Files/assets/73077301/95fea9c3-b8d1-4f71-a314-f0d5ebd81142

**Comment by Binekrasik** [comment]
> I believe you but I must be missing a step when trying to repro this, can you upload a screen recording (you can use the attach file feature in GitHub) so I can make sure I'm following the same steps?

Sorry for the delay, but I wasn't able to upload the video to Github because of the file size limit. I've uploaded it to streamable instead: https://streamable.com/6peo0v

**Comment by Josh65-2201** [comment]
@Binekrasik Is this still an issue on v4.0.11 https://files.community/download

**Comment by Binekrasik** [comment]
@Josh65-2201 Tried it just now using Files 4.0.11.0 on Windows 10.0.26100.4652 with the same executable, and it still launches the "zip" file just fine as it was an executable. I also tried to run it out of pure curiosity using powershell and command prompt, and while powershell opens the "open with..." popup thingy, cmd casually launches the executable just like Files did. I didn't search if this is the intended behavior, but perhaps it might be something to look into aswell.

<img width="1052" height="338" alt="Image" src="https://github.com/user-attachments/assets/07c0a45d-d764-4f64-a9cc-ef44af8dc751" />

**Comment by Josh65-2201** [comment]
File extensions are used to tell OS what program to open. Programs read the header in the file rather than the file extension to know what to do with it. That is probably why CMD opens it as an exe since that what the header says.

**Comment by yair100** [comment]
We already display a warning before changing file extensions. If someone goes ahead and renames zips to exes, I'm not sure we need to add custom logic to display a notification. As a sidenote, I believe we use PowerShell to launch executable files so this lines up with your findings.
